src/newcharacter.cpp

```diff
@@ -1063,19 +1075,17 @@ tab_direction set_traits( const catacurses::window &w, player &u, points_left &p
     const size_t iContentHeight = TERMY - 9;
     int iCurWorkingPage = 0;
 
-    int iStartPos[2];
-    iStartPos[0] = 0;
-    iStartPos[1] = 0;
-
-    int iCurrentLine[2];
-    iCurrentLine[0] = 0;
-    iCurrentLine[1] = 0;
+    int iStartPos[trait_pages];
+    int iCurrentLine[trait_pages];
+    size_t traits_size[trait_pages];
 
-    size_t traits_size[2];
-    traits_size[0] = vStartingTraits[0].size();
-    traits_size[1] = vStartingTraits[1].size();
+    for( int i = 0; i < trait_pages; i++ ) {
```

Comment:
You can initialize arrays directly like this: `int array[3] = { 1, 2, 3 };`.

And the size of the `vStartingTraits` vectors does not need to be stored separately. One can query the vector directly anytime. So instead of "caching" it here, call `vStartingTraits[...].size()` whenever you need it.